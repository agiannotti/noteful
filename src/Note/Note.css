.Note {
  --note-height: 150px;
  background-color: #806565;
  color: #cacaca;
  height: var(--note-height);
  margin-bottom: 8px;
  max-width: 450px;
  padding: 8px 0 8px 8px;
  position: relative;
}

.Note__delete {
  --note-button-height: 32px;
  background-color: #352c2c;
  border: 0;
  color: #bdbcbc;
  float: right;
  font-size: 12px;
  height: var(--note-button-height);
  position: relative;
  top: calc((var(--note-button-height) + 32px) * -1);
  transform-origin: right bottom;
  transform: rotate(-90deg);
  width: calc(var(--note-height));
}
.Note__delete:hover {
  background-color: #520c0c;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

.Note__title {
  font-family: var(--custom-font);
  font-size: 20px;
  margin: 0;
  max-width: 80%;
  min-height: 24px;
}
.Note__title a {
  color: #cacaca;
  display: block;
  text-decoration: none;
}
.Note__title a:hover {
  color: #ffb72f;
}

.Note__dates {
  bottom: 8px;
  color: #a7a5a5;
  font-size: 10px;
  position: absolute;
  width: 75%;
}
