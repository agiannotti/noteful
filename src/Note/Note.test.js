import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Note from './Note'

describe(`Note component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .Note by default', () => {
    const wrapper = shallow(<Note />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Note given props', () => {
    const wrapper = shallow(<Note {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
