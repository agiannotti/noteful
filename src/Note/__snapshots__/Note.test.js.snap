// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a .Note by default 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      replace={false}
      to="/note/undefined"
    />
  </h2>
  <button
    className="Note__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Note__dates"
  >
    <div
      className="Note__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        Invalid Date
      </span>
    </div>
  </div>
</div>
`;

exports[`Note component renders the Note given props 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      replace={false}
      to="/note/a"
    >
      test-class-name
    </Link>
  </h2>
  <button
    className="Note__delete"
    type="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    remove
  </button>
  <div
    className="Note__dates"
  >
    <div
      className="Note__dates-modified"
    >
      Modified
       
      <span
        className="Date"
      >
        15th Jan 2019
      </span>
    </div>
  </div>
</div>
`;
